### PR TITLE
Move from const enums to consts

### PR DESCRIPTION
**Pull request by wycats**
This PR ended up being much more extensive than originally intended because it ended up including quite a bit of cleanup to the benchmark setup and cleaned up mistakes in the `package.json`s that were causing subtle issues in the benchmark environment.

[EDIT] @NullVoxPopuli pushed me to extract a lot of the incidental changes into their own PRs. I did quite a bit of that and it did streamline this PR. Thanks @NullVoxPopuli!

### CODE REVIEW

File: packages/@glimmer-workspace/integration-tests/lib/setup-harness.ts
Comment (NullVoxPopuli) [inline packages/@glimmer-workspace/integration-tests/lib/setup-harness.ts:1]: Probably should have been a separate PR

File: bin/setup-bench.mjs
Comment (NullVoxPopuli) [inline bin/setup-bench.mjs:1]: Probably should have been a separate PR

File: packages/@glimmer/compiler/index.ts [hunk 1]
```diff
@@ -7,8 +7,9 @@ export {
   s,
   unicode,
 } from './lib/builder/builder';
-export { Builder, type BuilderStatement } from './lib/builder/builder-interface';
+export { type BuilderStatement } from './lib/builder/builder-interface';
 export { defaultId, precompile, precompileJSON, type PrecompileOptions } from './lib/compiler';
+export * from '@glimmer/constants';
```
Comment (NullVoxPopuli) [inline packages/@glimmer/compiler/index.ts:12]: How many things are in here?
Why *?

(This is an expansion of public api for glimmer compiler, so i just want to make sure this makes sense, rather than mostly a convenience (because idk))

File: packages/@glimmer/compiler/lib/passes/2-encoding/index.ts
Comment (NullVoxPopuli) [inline packages/@glimmer/compiler/lib/passes/2-encoding/index.ts:1]: These logging changes probably should have been a separate PR

File: packages/@glimmer/compiler/package.json
Comment (NullVoxPopuli) [inline packages/@glimmer/compiler/package.json:1]: Probably should have been a separate PR?

File: packages/@glimmer/compiler/test/compiler-test.ts [hunk 2]
```diff
@@ -50,8 +51,8 @@ function test(desc: string, template: string, ...expectedStatements: BuilderStat
   });
 }
 
-const Append = Builder.Append;
-const Concat = Builder.Concat;
+const Append = BUILDER_APPEND;
```
Comment (NullVoxPopuli) [inline packages/@glimmer/compiler/test/compiler-test.ts:54]: Why alias these?

It's how it was, and unaliasing them should probably be a separate PR

File: packages/@glimmer/debug-util/package.json
Comment (NullVoxPopuli) [inline packages/@glimmer/debug-util/package.json:1]: Probably should have been a separate PR

File: packages/@glimmer/local-debug-flags/index.ts [hunk 3]
```diff
@@ -1,23 +1,171 @@
...
-      return false;
+/* eslint-disable no-console */
+
+declare global {
+  interface ImportMetaEnv {
+    VM_LOCAL_DEV?: boolean;
+  }
+}
+
+// All of these flags are expected to become constant `false` in production builds.
```
Comment (NullVoxPopuli) [inline packages/@glimmer/local-debug-flags/index.ts:9]: And those that exist through to ember (via debug mode, which is always used) should be strippable through ember's build process.
And those that make it through that (hopefully just DEBUG at that point) should be strippable by userland terser

File: packages/@glimmer/local-debug-flags/index.ts
Comment (NullVoxPopuli) [inline packages/@glimmer/local-debug-flags/index.ts:1]: Probably should have been a separate PR 🙈

File: packages/@glimmer/runtime/lib/opcodes.ts [hunk 4]
```diff
@@ -190,4 +88,113 @@ export class AppendOpcodes {
   }
 }
 
+if (import.meta.env.VM_LOCAL_DEV) {
```
Comment (NullVoxPopuli) [inline packages/@glimmer/runtime/lib/opcodes.ts:91]: Nice!

**Review by NullVoxPopuli — APPROVED** [review]
Lots of changes! But overall looks good, and all the types of changes here are straight forward.

Good improvement, yay!

Left a question tho
(Doesn't affect approval)

File: packages/@glimmer/compiler/index.ts (hunk 1 above)
Comment (wycats) [reply packages/@glimmer/compiler/index.ts:12]: This is a good point. There are a few things that were re-exported before when they were a single enum *type* that I carried forward and shouldn't have.

File: packages/@glimmer/compiler/test/compiler-test.ts (hunk 2 above)
Comment (wycats) [reply packages/@glimmer/compiler/test/compiler-test.ts:54]: These are entirely internal aliases in this test file. It makes sense to unalias them, absolutely.

File: packages/@glimmer/local-debug-flags/index.ts
Comment (wycats) [reply packages/@glimmer/local-debug-flags/index.ts:1]: I addressed all of the other extraction requests. This one is sensible, but it would be a bunch of fine-grained work to pull it out and I'd rather just move ahead with this now that I pulled #1646, #1648 and 57d9118 into their own granular changes.

**Comment by wycats** [comment]
image

Local results are "no change". I would have expected at least a small change, but the main motivation for this change was to move away from enums and const enums in favor of normal constants and literal types, which eliminates a dependency on increasingly non-standard compilation infrastructure.

So no news is good news.

**Review by NullVoxPopuli — APPROVED** [review]

**Comment by NullVoxPopuli** [comment]
No perf difference on either of our local tests -- CI thinks improvement, so we'll merge.
